### PR TITLE
INT-4154: Merge Files Java DSL

### PR DESCRIPTION
**Pull request by artembilan**
JIRA: https://jira.spring.io/browse/INT-4154

* Move `Transformers` for files to the `Files` factory
* Remove `@Deprecated` methods
* Fix `FilesTests` for new packages and removed methods like `.handleWithAdapter()`

### CODE REVIEW

File: spring-integration-file/src/main/java/org/springframework/integration/file/dsl/FileTransferringMessageHandlerSpec.java [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+import org.springframework.integration.file.FileNameGenerator;
+import org.springframework.integration.file.remote.RemoteFileTemplate;
+import org.springframework.integration.file.remote.handler.FileTransferringMessageHandler;
+import org.springframework.integration.file.remote.session.SessionFactory;
+import org.springframework.integration.file.support.FileExistsMode;
+import org.springframework.messaging.Message;
+import org.springframework.util.Assert;
+
+/**
+ *
```
Comment (garyrussell) [inline spring-integration-file/src/main/java/org/springframework/integration/file/dsl/FileTransferringMessageHandlerSpec.java]: Javadoc

File: spring-integration-file/src/main/java/org/springframework/integration/file/dsl/FileWritingMessageHandlerSpec.java [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+import org.springframework.integration.file.DefaultFileNameGenerator;
+import org.springframework.integration.file.FileNameGenerator;
+import org.springframework.integration.file.FileWritingMessageHandler;
+import org.springframework.integration.file.support.FileExistsMode;
+import org.springframework.messaging.Message;
+import org.springframework.scheduling.TaskScheduler;
+import org.springframework.util.Assert;
+
+/**
+ * @author Artem Bilan
```
Comment (garyrussell) [inline spring-integration-file/src/main/java/org/springframework/integration/file/dsl/FileWritingMessageHandlerSpec.java]: Javadoc

File: spring-integration-file/src/main/java/org/springframework/integration/file/dsl/FileWritingMessageHandlerSpec.java [hunk 3]
```diff
@@ -0,0 +1,191 @@
...
+	public FileWritingMessageHandlerSpec charset(String charset) {
+		this.target.setCharset(charset);
+		return _this();
+	}
+
+	/**
+	 * @param appendNewLine true if a new-line should be written to the file after payload is written.
+	 * @return the spec.
+	 * @see FileWritingMessageHandler#setAppendNewLine(boolean)
+	 */
```
Comment (garyrussell) [inline spring-integration-file/src/main/java/org/springframework/integration/file/dsl/FileWritingMessageHandlerSpec.java]: We're a bit inconsistent here with some methods having Javadoc, others not (and none in the previous spec).

I am not saying we have to address it now, but we should make sure when we do docs for RC/GA that we addres these too.

File: spring-integration-file/src/main/java/org/springframework/integration/file/dsl/RemoteFileOutboundGatewaySpec.java [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+import org.springframework.integration.file.filters.CompositeFileListFilter;
+import org.springframework.integration.file.filters.FileListFilter;
+import org.springframework.integration.file.filters.RegexPatternFileListFilter;
+import org.springframework.integration.file.filters.SimplePatternFileListFilter;
+import org.springframework.integration.file.remote.gateway.AbstractRemoteFileOutboundGateway;
+import org.springframework.messaging.Message;
+import org.springframework.util.Assert;
+
+/**
+ *
```
Comment (garyrussell) [inline spring-integration-file/src/main/java/org/springframework/integration/file/dsl/RemoteFileOutboundGatewaySpec.java]: Javadoc

File: spring-integration-file/src/main/java/org/springframework/integration/file/dsl/RemoteFileOutboundGatewaySpec.java [hunk 5]
```diff
@@ -0,0 +1,175 @@
...
+
+	public <P> S localDirectory(Function<Message<P>, String> localDirectoryFunction) {
+		return localDirectoryExpression(new FunctionExpression<>(localDirectoryFunction));
+	}
+
+	/**
+	 * @param localDirectoryExpression a SpEL expression to evaluate the local directory.
+	 * @return the Spec.
+	 */
+	public S localDirectoryExpression(Expression localDirectoryExpression) {
```
Comment (garyrussell) [inline spring-integration-file/src/main/java/org/springframework/integration/file/dsl/RemoteFileOutboundGatewaySpec.java]: again, consistency

**Review by garyrussell — CHANGES_REQUESTED** [review]
We should at least address the class-level javadocs and at a minimum open a cleanup JIRA for the others.

File: spring-integration-file/src/main/java/org/springframework/integration/file/dsl/RemoteFileOutboundGatewaySpec.java (hunk 5 above)
Comment (artembilan) [inline spring-integration-file/src/main/java/org/springframework/integration/file/dsl/RemoteFileOutboundGatewaySpec.java]: Np. I'm on it.
Addressing all the missed JavaDocs in these DSL classes for File module...

**Comment by artembilan** [comment]
Doing this I noticed that we are missing `srteaming` adapters DSL.

If you wish I can provide provide a base `Spec` class here in the File `dsl` package or you can extract it during (S)FTP work.

Works for me any way.

**Comment by garyrussell** [comment]
I will take care of the streaming stuff while doing (S)FTP.

**Review by garyrussell — APPROVED** [review]

**Comment by garyrussell** [comment]
Merged as f71d6a0e66887eba75ad4dd634b3220c6aa74d0e

**Pull request closed without merging**